ratis-server/src/main/java/org/apache/ratis/server/impl/ServerState.java

```diff
@@ -389,6 +389,7 @@ void setRaftConf(RaftConfiguration conf) {
     if (!listeners.isEmpty()) {
       server.getServerRpc().addRaftPeers(listeners);
     }
+    server.checkAndUpdateListenerState().join();
```

Comment:
This setRaftConf(..) method is used by many cases including initRaftLog and leader appendConfiguration.  It is better to check listener in the updateConfiguration(..) method.

BTW, there is another bug that we should also change SnapshotInstallationHandler to call updateConfiguration(..) instead of setRaftConf(..).  Otherwise, the conf change will not be handled correctly.